In [3]:
import numpy as np
from scipy.stats import chi2_contingency, chi2

# Chi-Square Test of Independence


# Data Description:
# Mizzare Corporation has collected data to explore if there's a significant association
# between the type of smart home device purchased (Smart Thermostats vs. Smart Lights)
# and the customer satisfaction level.

# Hypotheses:
# Null Hypothesis (H0): There is no association between the type of device and customer satisfaction.
# Alternative Hypothesis (H1): There is an association between the type of device and customer satisfaction.

# Data Setup:
# The following matrix represents the observed frequencies of customer satisfaction levels
# for two types of devices: Smart Thermostats and Smart Lights.
data = np.array([
    [50, 70],  # Very Satisfied
    [80, 100], # Satisfied
    [60, 90],  # Neutral
    [30, 50],  # Unsatisfied
    [20, 50]   # Very Unsatisfied
])

# Performing the Chi-Square test
chi2_stat, p_val, dof, expected = chi2_contingency(data)

# Results:
print("Chi-Square Statistic:", chi2_stat)
print("Degrees of Freedom:", dof)
print("P-value:", p_val)
print("Expected Frequencies:")
print(expected)

# Critical Value Calculation:
# Using a significance level (alpha) of 0.05
alpha = 0.05
critical_value = chi2.ppf(1 - alpha, dof)
print("Critical Value:", critical_value)

# Decision Making:
if chi2_stat >= critical_value:
    print("Reject the null hypothesis - There is a significant association between device type and customer satisfaction.")
else:
    print("Fail to reject the null hypothesis - There is no significant association between device type and customer satisfaction.")



Chi-Square Statistic: 5.638227513227513
Degrees of Freedom: 4
P-value: 0.22784371130697179
Expected Frequencies:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]
Critical Value: 9.487729036781154
Fail to reject the null hypothesis - There is no significant association between device type and customer satisfaction.


 Conclusion:
 Based on the Chi-Square statistic of 5.638 and a critical value of 9.487 for 4 degrees of freedom,
 we fail to reject the null hypothesis at a 5% significance level. The p-value of 0.2278 also indicates
 that the observed difference in satisfaction could reasonably occur by chance under the null hypothesis.
 This analysis concludes that there is no significant association between the type of smart home device purchased
 and the levels of customer satisfaction.